File: leetcode_rust/src/problems/problem_121.rs
pub fn max_profit(prices: Vec<i32>) -> i32 {
    let mut min = prices[0];
    let mut target = min;
    for i in 1..prices.len() {
        if prices[i] < min {
            target -= min - prices[i];
            min = prices[i];
        } else if prices[i] > target {
            target = prices[i];
        }
    }

    target - min
}

File: LeetCode/LeetCode/Problems/1/Problem121.cs
public class Problem121
{
    public int MaxProfit(int[] prices)
    {
        var min = prices[0];
        var target = min;
        for (var i = 1; i < prices.Length; i++)
        {
            if (prices[i] < min)
            {
                target -= min - prices[i];
                min = prices[i];
            }
            else if (prices[i] > target)
            {
                target = prices[i];
            }
        }

        return target - min;
    }
}